src/Microsoft.Tye.Hosting/TyeHost.cs

```diff
@@ -204,6 +204,13 @@ private IHost BuildWebApplication(Application application, HostOptions options,
                                     });
                             });
                     services.AddSingleton(application);
+                    services.AddScoped(sp =>
```

Comment:
Not sure if this is necessary. You should be able to access `HttpContext.Request` from Razor pages/views: https://learn.microsoft.com/en-us/aspnet/core/fundamentals/http-context?view=aspnetcore-7.0#access-httpcontext-from-razor-pages and build the URL to the start/stop API request from that.